bridges/snowbridge/primitives/router/src/inbound/mod.rs

```diff
@@ -262,8 +262,15 @@ impl<
 			// Pay for execution.
 			BuyExecution { fees: xcm_fee, weight_limit: Unlimited },
 			// Fund the snowbridge sovereign with the required deposit for creation.
-			DepositAsset { assets: Definite(deposit.into()), beneficiary: bridge_location },
-			// Only our inbound-queue pallet is allowed to invoke `UniversalOrigin`
+			DepositAsset { assets: Definite(deposit.into()), beneficiary: bridge_location.clone() },
+			// This `SetAppendix` ensures that `xcm_fee` not spent by `Transact` will be
+			// deposited to snowbridge sovereign, instead of being trapped, regardless of
+			// `Transact` success or not.
+			SetAppendix(Xcm(vec![DepositAsset {
```

Comment:
If we put this one instruction before it can be executed in case there's an error in the previous `DepositAsset`. Although that shouldn't happen, it's a good idea to set this as early as possible

Reply:
That should not be possible in practice, but even if it were, the appendix is the same `DepositAsset` instruction, so it would fail in the appendix too.